### PR TITLE
implement `U256` Multiply/Divide; `full_math` fixes; `dydx_math` testing

### PR DESCRIPTION
**Pull request by alphak3y**

### CODE REVIEW

File: libraries/src/concentrated/SQ63x64.sw [hunk 1]
```diff
@@ -142,85 +142,80 @@ impl core::ops::Subtract for SQ63x64 {
             value: self.value - other.value,
         }
     }
-}
+} 
 impl SQ63x64 {
     /// Multiply a SQ63x64 with a SQ63x64. Panics of overflow.
     //TODO: assumes positive values
-    fn multiply(self, other: Self) -> Self {
+    pub fn multiply(self, other: Self) -> Self {
```
Comment (ControlCplusControlV) [inline libraries/src/concentrated/SQ63x64.sw]: Isn't this inherently public because its in a impl block not on a trait?

File: libraries/src/concentrated/SQ63x64.sw [hunk 2]
```diff
@@ -256,7 +251,7 @@ impl SQ63x64 {
         let half_scaling_unit = U128::from((0,1 << 63)) / two_u128;
         let mut delta = half_scaling_unit;
         // will perform 31 iterations
-        let zero = U128::from((0,1<<(62-31)));
+        let zero = U128::from((0,1<<(62-62)));
```
Comment (ControlCplusControlV) [inline libraries/src/concentrated/SQ63x64.sw]: Why was this changed?

File: libraries/src/concentrated/tick_math.sw [hunk 3]
```diff
@@ -444,41 +460,27 @@ pub fn get_price_sqrt_at_tick(tick: I24) -> Q64x64 {
...
-        value: U128 {
-            upper: 0,
-            lower: 0,
-        },
-    }
-}
-
 pub fn get_tick_at_price(sqrt_price: Q64x64) -> I24 {
-    check_sqrt_price_bounds(sqrt_price);
+    //return I24::from_uint(0);
```
Comment (ControlCplusControlV) [inline libraries/src/concentrated/tick_math.sw]: pls remove

File: libraries/src/concentrated/tick_math.sw [hunk 4]
```diff
@@ -444,41 +460,27 @@ pub fn get_price_sqrt_at_tick(tick: I24) -> Q64x64 {
...
+    //check_sqrt_price_bounds(sqrt_price);
 
+    //return I24::from_uint(0);
```
Comment (ControlCplusControlV) [inline libraries/src/concentrated/tick_math.sw]:
```suggestion

```

File: tests/harness.rs [hunk 5]
```diff
@@ -68,37 +69,56 @@ async fn sq63x64() {
...
-    let result = contract_instance.methods()
-        .test_abs_u128()
-            .tx_params(my_tx_params)
-            .call()
-            .await
-            .unwrap()
-            .value;
-
-    println!("{}", result.upper);
+    // let result = contract_instance.methods()
```
Comment (ControlCplusControlV) [inline tests/harness.rs]: pls remove uncommented code

**Review by ControlCplusControlV — CHANGES_REQUESTED** [review]

File: libraries/src/concentrated/SQ63x64.sw (hunk 1 above)
Comment (alphak3y) [reply libraries/src/concentrated/SQ63x64.sw]: Believe you are correct yes. We'll know for sure as we test more.

File: libraries/src/concentrated/SQ63x64.sw (hunk 2 above)
Comment (alphak3y) [reply libraries/src/concentrated/SQ63x64.sw]: Yeah I think 31 iterations is sufficient from what I've seen.

We'll be able to make a final decision once we can weigh gas numbers vs. precision.

File: libraries/src/concentrated/tick_math.sw (hunk 3 above)
Comment (alphak3y) [reply libraries/src/concentrated/tick_math.sw]: Done

File: tests/harness.rs (hunk 5 above)
Comment (alphak3y) [reply tests/harness.rs]: Done

**Review by ControlCplusControlV — APPROVED** [review]
